test/unit/mesh-gateway-serviceaccount.bats

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bats
+
+load _helpers
+
+@test "meshGateway/ServiceAccount: disabled by default" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/mesh-gateway-serviceaccount.yaml  \
+      . | tee /dev/stderr |
+      yq 'length > 0' | tee /dev/stderr)
+  [ "${actual}" = "false" ]
+}
+
+@test "meshGateway/ServiceAccount: enabled with meshGateway.enabled = true" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/mesh-gateway-serviceaccount.yaml  \
+      --set 'meshGateway.enabled=true' \
+      --set 'connectInject.enabled=true' \
+      --set 'client.grpc=true' \
+      . | tee /dev/stderr |
+      yq 'length > 0' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+}
+
```

Comment:
Should test that this is disabled with connectInject and/or client.grpc turned off.

Reply:
The whole chart will fail in that case